cpp/tests/strings/json_tests.cpp

```diff
@@ -967,4 +967,40 @@ TEST_F(JsonPathTests, EscapeSequences)
     // clang-format on
     CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*result, expected);
   }
+}
+
+TEST_F(JsonPathTests, MissingFieldsAsNulls)
+{
+  std::string input_string{
+    // clang-format off
+    "{"
+      "\"tup\":"
+      "["
+          "{\"id\":\"1\",\"array\":[1,2]},"
+          "{\"id\":\"2\"},"
+          "{\"id\":\"3\",\"array\":[3,4]},"
+          "{\"id\":\"4\", \"a\": {\"x\": \"5\", \"y\": \"6\"}}"
+      "]"
+    "}"
+    // clang-format on
+  };
+  auto do_test = [&input_string](auto const& json_path_string, auto const& expected_string) {
+    cudf::test::strings_column_wrapper input{input_string};
+
+    cudf::strings::get_json_object_options options;
+    options.set_missing_fields_as_nulls(true);
+
+    auto const result =
+      cudf::strings::get_json_object(cudf::strings_column_view(input), {json_path_string}, options);
+
+    // expect
+    cudf::test::strings_column_wrapper expected({expected_string}, {1});
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*result, expected);
```

Comment:
```suggestion
    auto const result =
      cudf::strings::get_json_object(cudf::strings_column_view(input), {json_path_string}, options);
    cudf::test::strings_column_wrapper expected({expected_string}, {1});

    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*result, expected);
```

Reply:
> > set_missing_fields_as_nulls = false is set by default so I figured the existing tests would cover that case. Should I also have cases that explicitly set the condition?
> 
> I think so. We don't know, and are not guanteed, if the existing tests have that covered or not.

Agreeing with @ttnghia's assessment. I meant to post this as well: It would be good to test `set_missing_fields_as_nulls = true` case alongside the `false`. It will underline the difference in behaviour.